python/ray/dag/tests/test_accelerated_dag.py

```diff
@@ -176,6 +202,79 @@ def f(x):
         dag.experimental_compile()
 
 
+def test_dag_fault_tolerance(ray_start_regular_shared):
+    actors = [Actor.remote(0, fail_after=100, sys_exit=False) for _ in range(4)]
+    with InputNode() as i:
+        out = [a.inc.bind(i) for a in actors]
+        dag = MultiOutputNode(out)
+
+    compiled_dag = dag.experimental_compile()
+
+    for i in range(99):
+        output_channels = compiled_dag.execute(1)
+        # TODO(swang): Replace with fake ObjectRef.
+        results = [chan.begin_read() for chan in output_channels]
+        assert results == [i + 1] * 4
+        for chan in output_channels:
+            chan.end_read()
+
+    with pytest.raises(ValueError):
+        for i in range(99):
+            output_channels = compiled_dag.execute(1)
+            for chan in output_channels:
+                chan.begin_read()
+            for chan in output_channels:
+                chan.end_read()
+
+
+def test_dag_fault_tolerance_sys_exit(ray_start_regular_shared):
+    actors = [Actor.remote(0, fail_after=100, sys_exit=True) for _ in range(4)]
+    with InputNode() as i:
+        out = [a.inc.bind(i) for a in actors]
+        dag = MultiOutputNode(out)
+
+    compiled_dag = dag.experimental_compile()
+
+    for i in range(99):
+        output_channels = compiled_dag.execute(1)
+        # TODO(swang): Replace with fake ObjectRef.
+        results = [chan.begin_read() for chan in output_channels]
+        assert results == [i + 1] * 4
+        for chan in output_channels:
+            chan.end_read()
+
+    with pytest.raises(RaySystemError, match="channel closed"):
+        for i in range(99):
+            output_channels = compiled_dag.execute(1)
+            for chan in output_channels:
+                chan.begin_read()
+            for chan in output_channels:
+                chan.end_read()
+
+
+def test_dag_teardown_while_running(ray_start_regular_shared):
+    a = Actor.remote(0)
+
+    with InputNode() as inp:
+        dag = a.sleep.bind(inp)
+
+    compiled_dag = dag.experimental_compile()
+    compiled_dag.execute(3)  # 3-second slow task running async
```

Comment:
It'd be good to check that you can still begin_read on the returned channel here after teardown has been called.

I believe it can return either the actual value or the exception?